src/Data/Unrestricted/Linear.hs

```diff
@@ -95,35 +96,35 @@ import qualified Unsafe.Linear as Unsafe
 
 
 
--- | @Unrestricted a@ represents unrestricted values of type @a@ in a linear
+-- | @Ur a@ represents unrestricted values of type @a@ in a linear
 -- context. The key idea is that because the contructor holds @a@ with a
--- regular arrow, a function that uses @Unrestricted a@ linearly can use @a@
+-- regular arrow, a function that uses @Ur a@ linearly can use @a@
 -- however it likes.
--- > someLinear :: Unrestricted a #-> (a,a)
--- > someLinear (Unrestricted a) = (a,a)
-data Unrestricted a where
-  Unrestricted :: a -> Unrestricted a
+-- > someLinear :: Ur a #-> (a,a)
+-- > someLinear (Ur a) = (a,a)
+data Ur a where
+  Ur :: a -> Ur a
 
--- | Get an @a@ out of an @Unrestricted a@. If you call this function on a
--- linearly bound @Unrestricted a@, then the @a@ you get out has to be used
+-- | Get an @a@ out of an @Ur a@. If you call this function on a
+-- linearly bound @Ur a@, then the @a@ you get out has to be used
 -- linearly, for example:
 --
--- > restricted :: Unrestricted a #-> b
--- > restricted x = f (unUnrestricted x)
+-- > restricted :: Ur a #-> b
+-- > restricted x = f (unUr x)
 -- >   where
 -- >     -- f __must__ be linear
 -- >     f :: a #-> b
 -- >     f x = ...
-unUnrestricted :: Unrestricted a #-> a
-unUnrestricted (Unrestricted a) = a
+unUr :: Ur a #-> a
```

Comment:
`Ur` is itself a contraction. So I'd even go for `unur` here. Simple easy-to-spell and type combinator.

Reply:
Cool :sunglasses: